fetch-in-react-practice/client/components/app.jsx

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import PageTitle from './page-title';
+import TodoList from './todo-list';
+import TodoForm from './todo-form';
+
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos: []
+    };
+    this.addTodo = this.addTodo.bind(this);
+    this.toggleCompleted = this.toggleCompleted.bind(this);
+  }
+
+  componentDidMount() {
+    /**
+     * Use fetch to send a GET request to `/api/todos`.
+     * Then 😉, once the response JSON is received and parsed,
+     * update state with the received todos.
+     */
+
+    fetch('/api/todos')
+      .then(response => response.json())
+      .then(data => this.setState({ todos: data }));
+  }
+
+  addTodo(newTodo) {
+    /**
+    * Use fetch to send a POST request to `/api/todos`.
+    * Then 😉, once the response JSON is received and parsed,
+    * add the created todo to the state array.
+    *
+    * Do not mutate the original state array, nor any objects within it.
+    * https://reactjs.org/docs/optimizing-performance.html#the-power-of-not-mutating-data
+    *
+    * TIP: Be sure to SERIALIZE the todo object in the body with JSON.stringify()
+    * and specify the "Content-Type" header as "application/json"
+    *
+    * TIP: Use Array.prototype.concat to create a new array containing the contents
+    * of the old array, plus the object returned by the server.
+    */
+
+    const requestOptions = {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({ task: newTodo.task })
+    };
+
+    fetch('/api/todos', requestOptions)
+      .then(response => response.json())
+      .then(data => this.setState({ todos: this.state.todos.concat(data) })
+      );
+  }
+
+  toggleCompleted(todoId) {
+    /**
+     * Find the index of the todo with the matching todoId in the state array.
+     * Get its "isCompleted" status.
+     * Make a new object containing the opposite "isCompleted" status.
+     * Use fetch to send a PATCH request to `/api/todos/${todoId}`
+     * Then 😉, once the response JSON is received and parsed,
+     * replace the old todo in the state array.
+     *
+     * NOTE: "toggle" means to flip back and forth, so clicking a todo
+     * in the list should "toggle" its isCompleted status back and forth.
+     *
+     * Do not mutate the original state array, nor any objects within it.
+     * https://reactjs.org/docs/optimizing-performance.html#the-power-of-not-mutating-data
+     *
+     * TIP: Be sure to SERIALIZE the updates in the body with JSON.stringify()
+     * And specify the "Content-Type" header as "application/json"
+     */
+
+    const requestOptions = {
+      method: 'PATCH',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({ isCompleted: true })
```

Comment:
You can't always assume it will need to be set to true, it should be able to toggle `isComplete` to either `true` or `false`. You need to find the correct item in the `todos` array and use its current `isCompleted` property at set it to the opposite to send to the server.